pkg/controller/dnszone/awsactuator.go

```diff
@@ -472,6 +472,53 @@ func (a *AWSActuator) Exists() (bool, error) {
 	return a.zoneID != nil, nil
 }
 
+// SetConditionsForError sets conditions on the dnszone given a specific error. Returns true if conditions changed.
+func (a *AWSActuator) SetConditionsForError(err error) bool {
+	awsErr, ok := err.(awserr.Error)
+	if !ok {
+		return false
+	}
+
+	conds := []hivev1.DNSZoneCondition{}
+	changed := false
+	failureMessage := ""
+
+	switch awsErr.Code() {
+	case "AccessDeniedException":
+		conds, changed = controllerutils.SetDNSZoneConditionWithChangeCheck(
+			a.dnsZone.Status.Conditions,
+			hivev1.InsufficientCredentialsCondition,
+			corev1.ConditionTrue,
+			accessDeniedReason,
+			awsErr.Message(),
+			controllerutils.UpdateConditionIfReasonOrMessageChange,
+		)
+		failureMessage = "Credentials are lacking permissions"
+
+	case "InvalidSignatureException":
+		fallthrough
+	case "UnrecognizedClientException":
```

Comment:
I think the following conveys the logic a bit better. The fallthough, at least to me, implies that the first case did something and should fall through to do more.
```suggestion
	case "InvalidSignatureException", "UnrecognizedClientException":
```